### ISSUE TITLE
Advanced web UI (hlky)

### ISSUE THREAD

**Issue by mchaker** [NickLucche/stable-diffusion-nvidia-docker#6]
I've tried using this web UI and it seems more featured: https://github.com/hlky/stable-diffusion-webui

Specifically the face fix and upscale models I was able to add and enable.

Is there a way I could use that web UI with your docker container release?

**Comment by NickLucche** [comment]
Hey, this is a really comprehensive project I was unaware of, thanks for sharing that.
I'm afraid that the simplest solution here would be to containerize their release, the UI seems to have many features that either depend on other models (upsampling) or add additional post-processing, it wouldn't be very wise to port that.
Perhaps you could open an issue on their repo about it.

**Comment by mchaker** [comment]
Sounds good, I'll look into containerizing (or asking them for containerizing) their UI :)

**Comment by angelmankel** [comment]
Would it be hard to set up endpoints to allow a react frontend for this project? I would be willing to make a React UI with some (or possibly all with some help) of those same features. I think having a proper image gallery where you can compare/recall/etc would be great for artists as well instead of the terrible gradio one. I just don't know flask/django to set up the backend part of it.

**Comment by NickLucche** [comment]
Yep I agree I think it could be a nice feature to have. I can setup a flask/django app for serving the .js files as well as a rest API for doing the same things you can do with gradio right now. 
I know very little about react tho so I wouldn't be able to help on that side.

**Comment by angelmankel** [comment]
That would be awesome if you could set that up. I could handle the React side of things. If you'd like, I can let you know when I have a good idea of how the UI should look/function before I actually do any coding. The community might have some input on that as well. If so, just let me know where I should do that exactly since this is a bit unrelated to the original issue opened here. (I'm not very familiar with Github best practices yet)

**Comment by NickLucche** [comment]
Sounds great, let me know when you have a mock-up you might want to share.
To do so, it would be best to open a new issue and discuss about the changes there. 
To add the code later on, you should fork this repo and make a new branch with the js files!

**Comment by NickLucche** [comment]
@angelmankel I've tested a simple flask server ([here](https://github.com/NickLucche/stable-diffusion-nvidia-docker/blob/rest-apis/api.py)) for serving a sample react app and it seems to be working. 
I've created a test app in the root folder using `npx create-react-app stable-diffusion-ui`.  

On the second thought, I think I can actually set up a separate repo for the react UIx that we can then link as a github submodule. You can find the sample react app here https://github.com/NickLucche/stable-diffusion-react-ui.

**Comment by angelmankel** [comment]
Wow that was fast. This is great, I will get back to you with a mock-up in the new repo soon! Thanks for getting this all started, really looking forward to working on this.
